### PR TITLE
update: Write certain errors to stderr

### PR DESCRIPTION
**Pull request by mrshu**
* As described and requested in #1202, this commit makes errors on
  command line syntax and files not found to go to stderr rather than
  stdout.

Signed-off-by: mr.Shu <mr@shu.io>

### CODE REVIEW

**Comment by ddfisher** [comment]
These changes look good to me, but it still prints file not found errors (e.g. `mypy: can't read file 'foo': No such file or directory`) to stdout.  Could you fix that up also?

File: mypy/errors.py [hunk 1]
```diff
@@ -370,12 +370,14 @@ def report_internal_error(err: Exception, file: str, line: int) -> None:
...
     tb2 = traceback.extract_tb(sys.exc_info()[2])
-    print('Traceback (most recent call last):')
+    print('Traceback (most recent call last):', file=sys.stderr)
     for s in traceback.format_list(tb + tb2):
-        print(s.rstrip('\n'))
-    print('{}: {}'.format(type(err).__name__, err))
-    print('\n*** INTERNAL ERROR ***')
+        print(s.rstrip('\n'), file=sys.stderr)
+    print('{}: {}'.format(type(err).__name__, err), file=sys.stderr)
+    print('\n*** INTERNAL ERROR ***', file=sys.stderr)
```
Comment (gvanrossum) [inline mypy/errors.py]: I'd argue that it is intentional that the traceback is written to stdout but the final two messages are written to stderr. Though I don't feel strongly.

File: mypy/lex.py [hunk 2]
```diff
@@ -880,7 +880,7 @@ def ignore_break(self) -> bool:
     # Lexically analyze a file and dump the tokens to stdout.
     import sys
     if len(sys.argv) != 2:
-        print('Usage: lex.py FILE')
+        print('Usage: lex.py FILE', file=sys.stderr)
```
Comment (gvanrossum) [inline mypy/lex.py]: You realize this is just a bit of manual test code right? :-)

**Comment by gvanrossum** [comment]
Pin? Are you going to update this PR?

File: mypy/lex.py (hunk 2 above)
Comment (mrshu) [inline mypy/lex.py]: @gvanrossum I do now but I still feel this deserves to be written to stderr =)

**Comment by mrshu** [comment]
Thanks for the comment @ddfisher, I believe it should be fixed now.

@gvanrossum I just did. Please do let me know if you have more comments or other feedback.

**Comment by gvanrossum** [comment]
David, if you're happy you can merge this.

On Thursday, March 3, 2016, Marek Šuppa <notifications@github.com> wrote:

> Thanks for the comment @ddfisher <https://github.com/ddfisher>, I believe
> it should be fixed now.
>
> @gvanrossum <https://github.com/gvanrossum> I just did. Please do let me
> know if you have more comments or other feedback.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/python/mypy/pull/1251#issuecomment-191962216>.
>


-- 
--Guido (mobile)

**Comment by ddfisher** [comment]
This looks good to me.  Thanks!

**Comment by gvanrossum** [comment]
This PR actually reverted the key part of #1090, which was intended to write "regular" messages to stdout instead of stderr, per #1051. I will re-fix that without reverting the other parts of this PR -- errors about files not found and the like will continue to go to stderr.

**Pull request merged**